### PR TITLE
Fix zero price simple failed to resolve as default

### PR DESCRIPTION
**Pull request by torreytsui**
### Description

When two simple products of a configurable have different prices, the lowest price simple product is expected to resolve as the default price showing on product detail page.

If one of the simple products has price=0, it will fail to resolve as default.

Notice that it is an edge case having price=0 product. It probably is still worth fixing.

### Fixed Issues (if relevant)

Can't find any related issues at the moment

### Manual testing scenarios

1. Set up a configurable with two simple products. One simple have price at £3, another at £0
2. The simple products with *LOWER* product id must be at £0
3. Go to product detail page of the configurable

Expected result: Default price rendered as £0
Actual result: Default price rendered as £3

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=16540) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @torreytsui. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by torreytsui** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @torreytsui. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @torreytsui, here is your [Magento instance](https://i-16540-2-2-develop.engcom.dev.magento.com).
Admin access: [https://i-16540-2-2-develop.engcom.dev.magento.com/admin](https://i-16540-2-2-develop.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by torreytsui** [comment]
Tested on the instance and somehow can't replicate. I have done a 3v4l demo, hopefully this can convince as a bug.

https://3v4l.org/Jk6lT

**Comment by VladimirZaets** [comment]
Hi @torreytsui,  thank you for the collaboration. The issue doesn't reproduce, but I agree that your fix is correct. 
This is an important functionality, please cover your changes with tests. 
Thanks

File: app/code/Magento/ConfigurableProduct/Test/Unit/Pricing/Price/ConfigurablePriceResolverTest.php [hunk 1]
```diff
@@ -81,8 +88,40 @@ public function testResolvePrice($expectedValue)
...
+                ],
+                $expectedPrice = 0.00,
+            ],
+            'Multiple variants at price 10, 0, 20, should return 0.00 (float)' => [
+                $variantPrices = [
+                    10,
+                    0,
+                    20,
+                ],
+                $expectedPrice = 0.00,
```
Comment (torreytsui) [inline app/code/Magento/ConfigurableProduct/Test/Unit/Pricing/Price/ConfigurablePriceResolverTest.php:123]: Hi @VladimirZaets I have refactored the unit test to be more meaningful as test cases. New test cases are added to reflect the bug. Does it make sense to you?

Comment (VladimirZaets) [reply app/code/Magento/ConfigurableProduct/Test/Unit/Pricing/Price/ConfigurablePriceResolverTest.php:123]: Hi @torreytsui, yes, it is ok, thanks!

**Comment by magento-engcom-team** [comment]
Hi @torreytsui. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.6. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**